pkg/controller/managedresources/controller.go

```diff
@@ -187,23 +187,25 @@ func (r *Reconciler) reconcile(mr *resourcesv1alpha1.ManagedResource, log logr.L
 				// look up scope of objects' kind to check, if we should default the namespace field
 				mapping, err := r.targetRESTMapper.RESTMapping(obj.GroupVersionKind().GroupKind(), obj.GroupVersionKind().Version)
 				if err != nil || mapping == nil {
-					if meta.IsNoMatchError(err) {
-						// Reset RESTMapper in case of cache misses.
-						r.targetRESTMapper.Reset()
-					}
-					log.Error(err, fmt.Sprintf("could not get rest mapping for %s '%s/%s'", obj.GetKind(), obj.GetNamespace(), obj.GetName()),
+					// Don't reset RESTMapper in case of cache misses. Most probably indicates, that the corresponding CRD is not yet applied.
```

Comment:
Can you explain why the reset is not needed anymore? I see it's also done below, but I don't have the context about the relation anymore.

Reply:
Sure :)
At this point, the rest mapper is used to figure out the scope of the to-be-applied-object. Before, we were defaulting the object's namespace to `default` if the resource is not a `Namespace`, which then was leading to many references in the object status like this:
```yaml
  - apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    name: foo
    namespace: default
```

The healthcheck controller later picks up those references to get the objects from the api server. Before using the cached client for that, it wasn't a problem to specify a namespace in a key for a non-namespaced object. But the cached client can't handle that, so I figured, we should properly check the objects scope before defaulting the namespace field.

The problem now is, that if you create a ManagedResource containing a CRD (e.g. VPA) and a lot of corresponding CRs, the target mapper is reset for each CR, if the CRD has not been created yet (e.g. `seed-bootstrap` applied for the first time). This will therefore lead to a lot of discovery requests, which could even be rate limited in extreme cases. That's why I am not resetting the rest mapper here anymore.
Requests to create such CRs will anyways fail at the first try, until the CRD has been accepted, then the rest mapper will be reset and in the next try, it will probably succeed.
But as I am writing this, I figured, I should probably reset the mapper only once per reconcile down below instead of for each cache miss for the exact same reason. 😄

Reply:
Alright, thank you for the detailed explanation! 
> But as I am writing this, I figured, I should probably reset the mapper only once per reconcile down below instead of for each cache miss for the exact same reason. 😄

Sounds good :)